.github/workflows/CI_pytest.yml

```diff
@@ -45,22 +46,24 @@ jobs:
         echo $CONDA
 
     - name: Check the setup (II)
-      if: matrix.os != 'windows-latest'
+      if: matrix.os != 'windows-latest' || (matrix.os != 'macos-latest' && matrix.python-version != '3.9')
```

Comment:
Is the idea to undo these changes down the line ? If yes, then I'm fine with the changes.  If not (i.e. these will stay long-term), then I'd be more in favor of not running a specific configuration at all to start with (rather than skip specific ones at each step).

Reply:
hmm I think I would keep them as long as we support Python 3.9 as I don't see any sign of the GH actions maintainers to bring back support for Python 3.9 on arm (see https://github.com/actions/setup-python/issues/696). But once Python 3.9 is end-of-life, we can safely remove the lines and the corresponding entry in the test matrix.

Reply:
Could be temporary though as one of the last contributions of the above issues mentions (https://github.blog/changelog/2024-04-01-macos-14-sonoma-is-generally-available-and-the-latest-macos-runner-image/)

Reply:
Then I would suggest to do something like [this](https://stackoverflow.com/questions/68994484/how-to-skip-a-configuration-of-a-matrix-with-github-actions):

```
  runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, macos-latest, windows-latest]
        python-version: ['3.9', '3.10', '3.11']
        exclude:
        # excludes python 3.9 on macOS-latest
        - os: macos-latest
           python-version: '3.9'
```

Unless you tried already ?

Reply:
ah yes, much better, will try!